xarray/tests/test_dataarray.py

```diff
@@ -23,6 +23,7 @@
     requires_scipy, source_ndarray)
 
 
+@pytest.mark.filterwarnings('ignore:The inplace argument')
```

Comment:
Can we try fixing these tests now instead? Otherwise we may be stuck with large clean-up later.

For testing deprecations, I find it  often helps to convert warnings into an error, e.g., by adding `raise Exception` on the line after issuing the warning.

Reply:
I thought we might want to keep those tests around while people are still using that code path.

Reply:
I forgot the main reason for doing this: to make sure that xarray doesn't rely upon the deprecated behavior internally.

But if you already checked that, then I suppose this is fine.

Reply:
Oh I haven't checked that closely but I can do that tonight. 
(my impression was that the tests were testing inplace behaviour, but I see that could be wrong)